File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/YGFloatOptional.java
package io.github.orioncraftmc.meditate.internal;

public class YGFloatOptional {
    private final float value_;

    public YGFloatOptional() {
        // TODO: qNaN
        value_ = Float.NaN;
    }

    public YGFloatOptional(float value) {
        this.value_ = value;
    }

    public final float unwrap() {
        return value_;
    }

    public final boolean isUndefined() {
        return Float.isNaN(value_);
    }

}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/internal/YGFloatOptional.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package app.cash.redwood.yoga.internal

import kotlin.jvm.JvmInline

@JvmInline
internal value class YGFloatOptional(val value: Float = Float.NaN) {

  fun unwrap(): Float {
    return value
  }

  fun isUndefined(): Boolean {
    return value.isNaN()
  }
}
